### PR TITLE
added all missing CEF Key Names for Event Producers based on CEF…

### PR DESCRIPTION
**Pull request by gergelyv78**
In readme.md a typo is corrected and added the new CommonEventFormatV25.pdf link, because the old one is unavailable now.
The missing keys were added to the module and tests were updated as well.
Keys sorted alphabetically.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,4 +25,4 @@ Currently, the library only implements a subset of the permissable CEF
 extensions that were personally useful. I'm very happy to take PRs to extend
 coverage to the fully valid CEF extension set.
 
-[CEF]: https://www.protect724.hpe.com/servlet/JiveServlet/downloadBody/1072-102-9-20354/CommonEventFormatv23.pdf
+[CEF]: https://community.microfocus.com/dcvta86296/attachments/dcvta86296/connector-documentation/1197/2/CommonEventFormatV25.pdf
```
Comment (graingert) [inline README.md]: How about

https://web.archive.org/web/20190926101816/https://community.microfocus.com/dcvta86296/attachments/dcvta86296/connector-documentation/1197/2/CommonEventFormatV25.pdf

so it doesn't disappear again?

File: format_cef/cef.py [hunk 2]
```diff
@@ -125,15 +125,60 @@ def sanitise(t, debug_name):
...
 ipv4_addr = str_sanitiser(r'\.'.join(['\d{1,3}'] * 4))
+ipv6_addr = str_sanitiser(r'\:'.join(['[0-9a-fA-F]{1,4}'] * 8))  # only complete ipv6 address accepted
+mac_addr = str_sanitiser(r'\:'.join(['[0-9a-fA-F]{2}'] * 6))
 str_31 = str_sanitiser(max_len=31)
 str_63 = str_sanitiser(max_len=63)
+str_128 = str_sanitiser(max_len=128)
+str_255 = str_sanitiser(max_len=255)
 str_1023 = str_sanitiser(max_len=1023)
 
 # An incomplete list of valid CEF extensions
```
Comment (graingert) [inline format_cef/cef.py]: is this comment still valid?

**Comment by GMaxera** [comment]
The version of the package should be increased to reflect these changes

**Comment by graingert** [comment]
@GMaxera PRs shouldn't bump the version

**Comment by graingert** [comment]
@gergelyv78 can you rebase on `master` so you run the tests?

File: format_cef/cef.py [hunk 3]
```diff
@@ -125,15 +125,59 @@ def sanitise(t, debug_name):
...
 str_31 = str_sanitiser(max_len=31)
 str_63 = str_sanitiser(max_len=63)
+str_128 = str_sanitiser(max_len=128)
+str_255 = str_sanitiser(max_len=255)
 str_1023 = str_sanitiser(max_len=1023)
 
-# An incomplete list of valid CEF extensions
 valid_extensions = {
     'applicationProtocol': Extension('app', str_31),
+    'baseEventCount': Extension('cnt', int_sanitiser()),
```
Comment (graingert) [inline format_cef/cef.py]: @gergelyv78 did you manually write out this table by eye, or is there a machine readable copy of the spec somewhere?

Comment (gergelyv78) [reply format_cef/cef.py]: I did it by eye. Although I tried to double check the list, but there is a chance that a few misspell were made.
Even the guide looks failing in standardization, as the CEF Key Names mostly begins with small capitals except "C6a4Label" or "DevicePayloadId". I'm not sure if I followed standard all the time.

Comment (gergelyv78) [reply format_cef/cef.py]: Updated all the exceptions: "C6a4Label", "DevicePayloadId" and "DeviceOutboundInterface"

**Comment by gergelyv78** [comment]
One more note:
You can find the key "start" twice in the code. According to V25 the full name of this field is "startTime", so it was added. Due to compatibility reason, I left the other "start" in the code (with full name of "start"). I think they can live together.

**Comment by graingert** [comment]
Can you clean up your commits? There's a few spurious merges and some of your commits are authored by @gergelyv78 and some are authored by @xxxxxxxxxxxxx

**Comment by gergelyv78** [comment]
> Can you clean up your commits with a rebase? There's a few spurious merges and some of your commits are authored by @gergelyv78 and some are authored by @xxxxxxxxxxxxx

It must be ok now.

File: README.md [hunk 4]
```diff
@@ -25,4 +25,4 @@ Currently, the library only implements a subset of the permissable CEF
 extensions that were personally useful. I'm very happy to take PRs to extend
 coverage to the fully valid CEF extension set.
 
-[CEF]: https://www.protect724.hpe.com/servlet/JiveServlet/downloadBody/1072-102-9-20354/CommonEventFormatv23.pdf
+[CEF]: https://web.archive.org/web/20190926101816/https://community.microfocus.com/dcvta86296/attachments/dcvta86296/connector-documentation/1197/2/CommonEventFormatV25.pdf
```
Comment (graingert) [inline README.md]:
```suggestion
[CEF]: https://web.archive.org/web/20190930102530/https://community.microfocus.com/dcvta86296/attachments/dcvta86296/connector-documentation/1197/2/CommonEventFormatV25.pdf
```

weirdly that old archive link doesn't work anymore

Comment (gergelyv78) [reply README.md]: I hope it is now saved by the Wayback Machine.
Do you think, attaching this guide to the project would be useful? Or it's not a good idea?

**Pull request merged**